msal4j-sdk/src/main/java/com/microsoft/aad/msal4j/AbstractApplicationBase.java

```diff
@@ -347,7 +356,7 @@ private static Authority createDefaultAADAuthority() {
         serviceBundle = new ServiceBundle(
                 builder.executorService,
                 builder.httpClient == null ?
-                        new DefaultHttpClient(builder.proxy, builder.sslSocketFactory, builder.connectTimeoutForDefaultHttpClient, builder.readTimeoutForDefaultHttpClient) :
+                        determineHttpClientType(builder) :
                         builder.httpClient,
```

Comment:
Is builder.httpClient the http client passed in by the user? If so, this would mean that we will skip cert validation when a httpClient is passed in by the user?

Reply:
Yes, essentially there would be three varieties of HTTP client: a non-MSI default with no cert validation, an MSI default with cert validation, and an app developer's extension of the non-MSI default, which they can add cert validation to.

That design was my understanding from the discussion in [.NET's thread for this](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/4462): @bgavrilMS said MSAL can "use it's own HttpClient that doesn't go through the extensibility pipeline", i.e. this new cert validation shouldn't be part of the default HttpClient that a user could extend. 

(@bgavrilMS feel free to correct me if I got that backwards, if so I can just move it to the HttpClient interface and give app developers access to it by default in their implementations)

Reply:
Let me try to clarify @Avery-Dunn 

- MSI flows should use either user configured HttpClient or a default HttpClient
- For ServiceFabric and only for ServiceFabric, it is acceptable to use a different HttpClient which perfoms extra validation of the SSL cert. In this case the user configured HttpClient can be ignored. 
- Ideally the design should allow for unit tests to continue to work.